src/System.IO.Compression/tests/Performance/Perf.DeflateStream.cs

```diff
@@ -10,75 +10,70 @@ namespace System.IO.Compression.Tests
 {
     public class Perf_DeflateStream
     {
-        private static List<object[]> _compressedFiles;
-        private static List<object[]> _byteArraysToCompress;
-
-        /// <summary>
-        /// Yields an Enumerable list of paths to GZTestData files
-        /// </summary>
-        public static IEnumerable<object[]> CompressedFiles()
+        /// <returns></returns>
+        private static string CreateCompressedFile(CompressionType type)
         {
-            if (_compressedFiles == null)
-            {
-                PerfUtils utils = new PerfUtils();
-                _compressedFiles = new List<object[]>();
-                // Crypto random data
-                byte[] bytes = new byte[100000000];
-                var rand = RandomNumberGenerator.Create();
-                rand.GetBytes(bytes);
-                string filePath = utils.GetTestFilePath() + ".gz";
-                using (FileStream output = File.Create(filePath))
-                using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
-                    zip.Write(bytes, 0, bytes.Length);
-                _compressedFiles.Add(new object[] { filePath });
-
-                // Create a compressed file with repeated segments
-                bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(100000));
-                filePath = utils.GetTestFilePath() + ".gz";
-                using (FileStream output = File.Create(filePath))
-                using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
-                    for (int i = 0; i < 1000; i++)
+            byte[] bytes;
+            const int fileSize = 100000000;
+            PerfUtils utils = new PerfUtils();
+            string filePath = utils.GetTestFilePath() + ".gz";
+            switch (type)
+            {    
+                case CompressionType.CryptoRandom:
+                    bytes = new byte[fileSize];
+                    var rand = RandomNumberGenerator.Create();
+                    rand.GetBytes(bytes);
+                    using (FileStream output = File.Create(filePath))
+                    using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
+                        zip.Write(bytes, 0, bytes.Length);
+                    break;
+                case CompressionType.RepeatedSegments:
+                    bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(fileSize / 1000));
+                    using (FileStream output = File.Create(filePath))
+                    using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
+                        for (int i = 0; i < 1000; i++)
```

Comment:
I don't remember what the compression window size is, but you might want to check to see whether the algorithm will actually see these as repeated segments, or whether the segments are too long for it to see the pattern.